backend/curation/api/v1/curation/collections/common.py

```diff
@@ -195,11 +195,19 @@ def reshape_datasets_for_curation_api(
     as_version: bool = False,
     is_published: bool = False,
 ) -> List[dict]:
+    business_logic = get_business_logic()
     active_datasets = []
+    dataset_version_ids = []
+    dataset_versions = []
     for dv in datasets:
-        dataset_version = get_business_logic().get_dataset_version(dv) if isinstance(dv, DatasetVersionId) else dv
+        if isinstance(dv, DatasetVersion):
+            dataset_versions.append(dv)
+        else:
+            dataset_version_ids.append(dv)
+    dataset_versions.extend(business_logic.database_provider.get_dataset_versions_by_id(dataset_version_ids))
```

Comment:
we should add an empty list check, to avoid opening a session / querying for nothing in case all are DatasetVersion instances